service/src/main/java/org/cbioportal/service/ClinicalDataService.java

```diff
@@ -43,11 +42,10 @@ BaseMeta fetchMetaClinicalDataInStudy(String studyId, List<String> ids, List<Str
                                           String clinicalDataType) throws StudyNotFoundException;
     
     List<ClinicalData> fetchClinicalData(List<String> studyIds, List<String> ids, List<String> attributeIds,
-                                                          String clinicalDataType, String projection);
+                                                          String clinicalDataType, String projection, Boolean includeSample);
```

Comment:
instead of making a change to the mapper (join sample table with patient table), I would suggest to make a consecutive call to fetch sample information from the database.